### ISSUE TITLE
Can't get racer install via code or console

### ISSUE THREAD

**Issue by janus** [racer-rust/racer#802]
I tried install racer using github repo but it usually fails. I figured out that some of the dependencies might be out of versions. I therefore changed their version. with that I was able to download and compile the dependencies, however a different problem appeared . When is below. Here is my  Cargo.toml::

```toml
[dependencies]
log = "0.3.6"
syntex_syntax = "0.59.1"
syntex_errors = "0.59.1"
toml = { git = "https://github.com/alexcrichton/toml-rs.git", rev = "a626c36" }
env_logger = "0.4.3"
typed-arena = { git = "https://github.com/SimonSapin/rust-typed-arena.git" }
clap = { git = "https://github.com/kbknapp/clap-rs.git" }
```

```
error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:103:6
    |
103 | impl visit::Visitor for UseVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:147:6
    |
147 | impl visit::Visitor for PatBindVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:184:6
    |
184 | impl visit::Visitor for PatVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:340:14
    |
340 | impl<'c, 's> visit::Visitor for LetTypeVisitor<'c, 's> {
    |              ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:389:14
    |
389 | impl<'c, 's> visit::Visitor for MatchTypeVisitor<'c, 's> {
    |              ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:519:14
    |
519 | impl<'c, 's> visit::Visitor for ExprTypeVisitor<'c, 's> {
    |              ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:842:6
    |
842 | impl visit::Visitor for StructVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:869:6
    |
869 | impl visit::Visitor for TypeVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:902:6
    |
902 | impl visit::Visitor for TraitVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:916:6
    |
916 | impl visit::Visitor for ImplVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:943:6
    |
943 | impl visit::Visitor for ModVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:956:6
    |
956 | impl visit::Visitor for ExternCrateVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
   --> src/racer/ast.rs:978:6
    |
978 | impl visit::Visitor for GenericsVisitor {
    |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
    --> src/racer/ast.rs:1005:6
     |
1005 | impl visit::Visitor for EnumVisitor {
     |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
    --> src/racer/ast.rs:1184:6
     |
1184 | impl visit::Visitor for FnOutputVisitor {
     |      ^^^^^^^^^^^^^^ expected lifetime parameter

error[E0106]: missing lifetime specifier
    --> src/racer/ast.rs:1200:14
     |
1200 | impl<'c, 's> visit::Visitor for FnArgTypeVisitor<'c, 's> {
     |              ^^^^^^^^^^^^^^ expected lifetime parameter

error: aborting due to 16 previous errors

error: Could not compile `racer`.
```

When I follow vscode missing tools , I got the below:

```
    Updating registry `https://github.com/rust-lang/crates.io-index`
error: failed to parse manifest at `/home/dookl/.cargo/registry/src/github.com-1ecc6299db9ec823/racer-2.0.10/Cargo.toml`

Caused by:
  no `package` or `project` section found.
```

**Comment by janus** [comment]
I later managed to get vscode to start of compiling but I still got error message:

`    Updating registry `https://github.com/rust-lang/crates.io-index`
  Installing racer v2.0.10
 Downloading regex v0.1.80
 Downloading regex-syntax v0.3.9
 Downloading thread_local v0.2.7
 Downloading utf8-ranges v0.1.3
 Downloading aho-corasick v0.5.3
 Downloading memchr v0.1.11
 Downloading thread-id v2.0.0
 Downloading kernel32-sys v0.2.2
 Downloading winapi v0.2.8
 Downloading winapi-build v0.1.1
   Compiling bitflags v0.9.1
   Compiling unicode-width v0.1.4
   Compiling winapi v0.2.8
   Compiling winapi-build v0.1.1
   Compiling libc v0.2.32
   Compiling unicode-xid v0.0.3
   Compiling vec_map v0.8.0
   Compiling regex-syntax v0.3.9
   Compiling typed-arena v1.3.0
   Compiling utf8-ranges v0.1.3
   Compiling term v0.4.6
   Compiling strsim v0.6.0
   Compiling log v0.3.8
   Compiling bitflags v0.7.0
   Compiling ansi_term v0.9.0
   Compiling rustc-serialize v0.3.24
   Compiling kernel32-sys v0.2.2
   Compiling atty v0.2.3
   Compiling memchr v0.1.11
   Compiling term_size v0.3.0
   Compiling aho-corasick v0.5.3
   Compiling textwrap v0.8.0
   Compiling thread-id v2.0.0
   Compiling thread_local v0.2.7
   Compiling clap v2.26.2
   Compiling toml v0.2.1
   Compiling syntex_pos v0.52.0
   Compiling syntex_errors v0.52.0
   Compiling syntex_syntax v0.52.0
   Compiling regex v0.1.80
   Compiling env_logger v0.3.5
   Compiling racer v2.0.10
error[E0601]: main function not found

error: aborting due to previous error

error: failed to compile `racer v2.0.10`, intermediate artifacts can be found at `/tmp/cargo-install.0RdQrzlon1I0`

Caused by:
  Could not compile `racer`.`

**Comment by janus** [comment]
When I included --verbose  I found that it runs racer to times, first as lib and as main.rs.. and it is when it runs main.rs that it spits out error message

` Running `rustc --crate-name racer /home/emeka/.cargo/registry/src/github.com-1ecc6299db9ec823/racer-2.0.10/src/racer/lib.rs --crate-type lib --emit=dep-info,link -C opt-level=3 -C debuginfo=2 -C metadata=b774193352aaa202 -C extra-filename=-b774193352aaa202 --out-dir /tmp/cargo-install.woai6ZvDC26z/release/deps -L dependency=/tmp/cargo-install.woai6ZvDC26z/release/deps --extern env_logger=/tmp/cargo-install.woai6ZvDC26z/release/deps/libenv_logger-8a23d9966a8e0bea.rlib --extern syntex_errors=/tmp/cargo-install.woai6ZvDC26z/release/deps/libsyntex_errors-287e3b18a057f0cc.rlib --extern typed_arena=/tmp/cargo-install.woai6ZvDC26z/release/deps/libtyped_arena-748e0f4db2cfc5bd.rlib --extern syntex_syntax=/tmp/cargo-install.woai6ZvDC26z/release/deps/libsyntex_syntax-b71dc08a26532e86.rlib --extern log=/tmp/cargo-install.woai6ZvDC26z/release/deps/liblog-87e60b98b0770277.rlib --extern clap=/tmp/cargo-install.woai6ZvDC26z/release/deps/libclap-7e12b7cbe33c8d9c.rlib --extern toml=/tmp/cargo-install.woai6ZvDC26z/release/deps/libtoml-c3c9204cb8b366b0.rlib --cap-lintsallow`
     Running `rustc --crate-name racer /home/emeka/.cargo/registry/src/github.com-1ecc6299db9ec823/racer-2.0.10/src/bin/main.rs --crate-type bin --emit=dep-info,link -C opt-level=3 -C debuginfo=2 -C metadata=9a336b30df7a0202 -C extra-filename=-9a336b30df7a0202 --out-dir /tmp/cargo-install.woai6ZvDC26z/release/deps -L dependency=/tmp/cargo-install.woai6ZvDC26z/release/deps --extern env_logger=/tmp/cargo-install.woai6ZvDC26z/release/deps/libenv_logger-8a23d9966a8e0bea.rlib--extern syntex_errors=/tmp/cargo-install.woai6ZvDC26z/release/deps/libsyntex_errors-287e3b18a057f0cc.rlib --extern typed_arena=/tmp/cargo-install.woai6ZvDC26z/release/deps/libtyped_arena-748e0f4db2cfc5bd.rlib --extern syntex_syntax=/tmp/cargo-install.woai6ZvDC26z/release/deps/libsyntex_syntax-b71dc08a26532e86.rlib --extern log=/tmp/cargo-install.woai6ZvDC26z/release/deps/liblog-87e60b98b0770277.rlib --extern clap=/tmp/cargo-install.woai6ZvDC26z/release/deps/libclap-7e12b7cbe33c8d9c.rlib --extern toml=/tmp/cargo-install.woai6ZvDC26z/release/deps/libtoml-c3c9204cb8b366b0.rlib --extern racer=/tmp/cargo-install.woai6ZvDC26z/release/deps/libracer-b774193352aaa202.rlib --cap-lints allow`
error[E0601]: main function not found

error: aborting due to previous error

error: failed to compile `racer v2.0.10`, intermediate artifacts can be found at `/tmp/cargo-install.woai6ZvDC26z`

Caused by:
  Could not compile `racer`.

Caused by:
  process didn't exit successfully: `rustc --crate-name racer /home/emeka/.cargo/registry/src/github.com-1ecc6299db9ec823/racer-2.0.10/src/bin/main.rs --crate-type bin --emit=dep-info,link -C opt-level=3 -C debuginfo=2 -C metadata=9a336b30df7a0202 -C extra-filename=-9a336b30df7a0202 --out-dir /tmp/cargo-install.woai6ZvDC26z/release/deps -L dependency=/tmp/cargo-install.woai6ZvDC26z/release/deps --extern env_logger=/tmp/cargo-install.woai6ZvDC26z/release/deps/libenv_logger-8a23d9966a8e0bea.rlib --extern syntex_errors=/tmp/cargo-install.woai6ZvDC26z/release/deps/libsyntex_errors-287e3b18a057f0cc.rlib --extern typed_arena=/tmp/cargo-install.woai6ZvDC26z/release/deps/libtyped_arena-748e0f4db2cfc5bd.rlib --extern syntex_syntax=/tmp/cargo-install.woai6ZvDC26z/release/deps/libsyntex_syntax-b71dc08a26532e86.rlib --extern log=/tmp/cargo-install.woai6ZvDC26z/release/deps/liblog-87e60b98b0770277.rlib --extern clap=/tmp/cargo-install.woai6ZvDC26z/release/deps/libclap-7e12b7cbe33c8d9c.rlib --extern toml=/tmp/cargo-install.woai6ZvDC26z/release/deps/libtoml-c3c9204cb8b366b0.rlib --extern racer=/tmp/cargo-install.woai6ZvDC26z/release/deps/libracer-b774193352aaa202.rlib --cap-lints allow` (exit code: 101)`

**Comment by kngwyu** [comment]
Closed via #854
